torch/testing/_internal/common_methods_invocations.py

```diff
@@ -345,6 +345,58 @@ def __init__(self,
         #   outside a function's domain.
         self._domain_eps = 1e-5
 
+
+# Metadata class for binary "universal functions (ufuncs)" that accept two
+# tensor and have common properties
+class _BinaryUfuncInfo(OpInfo):
+    """Operator information for 'universal binary functions (binary ufuncs).'
+    These are functions of two tensors with common properties like:
+      - they are elementwise functions
+      - the output shape is determined by the input shape
+      - they typically have method and inplace variants
+      - they typically support the out kwarg
+      - they typically have NumPy or SciPy references
+    See NumPy's universal function documentation
+    (https://numpy.org/doc/stable/reference/ufuncs.html) for more details
+    about the concept of ufuncs.
+    """
+
+    def __init__(self,
+                 name,  # the string name of the function
+                 *,
+                 ref,  # a reference function
+                 dtypes=floating_types(),
+                 dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
+                 dtypesIfCUDA=floating_and_complex_types_and(torch.half),
+                 dtypesIfROCM=floating_types_and(torch.half),
+                 domain=(None, None),  # the [low, high) domain of the function
+                 handles_large_floats=True,  # whether the op correctly handles large float values (like 1e20)
```

Comment:
We can probably get away without adding metadata like `handles_large_floats`, `handles_extremals`, `handles_complex_extremals` and `supports_complex_to_float`. 

(In fact we should probably remove all these from the UnaryUfuncInfos, too.)